integration_test/README.md

```diff
@@ -142,7 +142,60 @@ Then, inside `applications/<application>/`:
 1.  (if necessary) `exercise`. This is only needed
     sometimes, e.g. to get the application to log to a particular file.
 1.  (if you want to test logging) `expected_logs.yaml`
-1.  (if you want to test metrics) `metric_name.txt`
+1.  (if you want to test metrics) `expected_metrics.yaml`
+
+### expected_metrics.yaml
+
+We use `expected_metrics.yaml` both as a test artifact and as a source for documentation. All metrics ingested from the integration should be documented here.
+
+A sample `expected_metrics.yaml` snippet looks like:
+
+```yaml
+- type: workload.googleapis.com/apache.current_connections
+  value_type: INT64
+  kind: GAUGE
+  monitored_resource: gce_instance
+  labels:
+    server_name: .*
+  representative: true
+  optional: true
+```
+
+`type`, `value_type` and `kind` come directly from the metric descriptor for that metric. `monitored_resource` should always be `gce_instance`.
+
+`labels` is an exhaustive list of labels associated with the metric. Each key in `labels` is the label name, and its value is a regular expression. During the test, each label returned by the time series for that metric is checked against `labels`: every label in the time series must be present in `labels`, and its value must match the regular expression.
+
+For example, if a metric defines a label `operation` whose values can only be `read` or `write`, then an appropriate `labels` map in `expected_metrics.yaml` would be as follows:
+
+```yaml
+  labels:
+    operation: read|write
+```
+
+Exactly one metric from each integration's `expected_metrics.yaml` must have `representative: true`. This metric can be used to detect when the integration is enabled.
+
+With `optional: true`, the test will not fail if the metric is missing. This can be useful for metrics that are not guaranteed to be present during the test, for example due to platform differences or unimplemented test setup procedures.
+
+`expected_metrics.yaml` can be generated using `generate_expected_metrics.go`:
```

Comment:
I picture the first.. It’s not ideal though but it’s easy to get them started this way. 

Without this script, it will be a guess game for blueMedora team to make the test pass. They will need to guess what valueType what labels that timeseries might have. As they don’t have access to our testing project, they have no where to query the metricsDescriptor or timesSeries. I think they their own testing project? With this assumption, whenever they develop a new integration, they probably try and test in their own project. Running this script against their own project, it will generate the expected metrics for them which they will use to compose the ops agent side PR

Reply:
I can see the generator being useful for new integrations that have lots of metrics, since the hand-writing can be tedious. But there certainly is a balance to be struck with source of truth.

Maybe there's a compromise here: have ObserveIQ include expected_metrics.yaml in the scoping docs, and however they come up with that text is up to them -- write by hand or generate automatically. Either way, it ends up in the same place in the scoping doc which lets us validate them against the metrics table as a source of truth. Then we can remove this script from the repo and just hand it to them as one-off reference material rather than code that needs to be versioned and maintained.

Reply:
I cut off the words that only three of us knows.. okay here it is: We agreed that in the first version we would let contributors use this script, as much effort has been taken in writing this script. We shall rewrite the script in some future point to use the otel side metrics.yaml as the starting point. 🙂

Reply:
Right -- the intent is for this to be a stopgap for contributors to not have to hand-write everything if they've already been testing in their own GCP projects. This whole script can be deprecated/decommissioned/removed once we come up with a plan for using the OTel metadata instead, which is not a trivial problem to solve.